tests/Unit/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -161,5 +161,358 @@ auto remove_nth(const std::tuple<ValueTypes...>& tup) noexcept {
       tup, std::make_index_sequence<N>(),
       std::make_index_sequence<sizeof...(ValueTypes) - N - 1>());
 }
+
+/// \ingroup TestingFrameworkGroup
+/// \brief test construction and assignment of a `VectorType` with a `ValType`
+template <typename VectorType, typename ValType>
+void vector_test_construct_and_assign(
+    typename tt::get_fundamental_type_t<ValType> low =
+        typename tt::get_fundamental_type_t<ValType>{-100},
+    typename tt::get_fundamental_type_t<ValType> high =
+        typename tt::get_fundamental_type_t<ValType>{100}) noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t size = sdist(gen);
+
+  VectorType size_constructed{size};
+  CHECK(size_constructed.size() == size);
+  auto generated_val1 = make_with_random_values<ValType>(make_not_null(&gen),
+                                                         make_not_null(&dist));
+
+  VectorType value_size_constructed{size, generated_val1};
+  CHECK(value_size_constructed.size() == size);
+  std::for_each(
+      value_size_constructed.begin(), value_size_constructed.end(),
+      [generated_val1](typename VectorType::value_type element) noexcept {
+        CHECK(element == generated_val1);
```

Comment:
Could you `CAPTURE_PRECISE(value_size_constructed);` in the lambda so the whole container gets printed? This is why the loop was explicit before